lib/virtual-net/src/lib.rs

```diff
@@ -7,6 +7,7 @@ pub mod composite;
 pub mod host;
 pub mod loopback;
 pub mod meta;
+pub mod ruleset;
 #[cfg(feature = "remote")]
```

Comment:
Consider adding a `FirewallNetworking<T>` which applies rules. 
This is optional.

Note: one annoyance is that you would also have to add wrapping socket implementations.

Reply:
As discussed, we decided to not do this for now since methods like `try_accept` cannot be protected from outside and we need the inner network to actually run and accept the connection which makes enforcing the ruleset a bit weird and dependent on the implementation details of the inner network.